### PR TITLE
New Version ProtoDC2

### PR DESCRIPTION
**Pull request by dkorytov**
The reader now checks the version of the catalog and adds/changes quantity_modifiers appropriately. 
Ellipticity quantities are exposed. 

The _get_quantity_info_dict is mostly implemented. There is some bug though at the moment when looking to subgroup, eg: 'x' works fine, 'morphology/inclination' does not.

### CODE REVIEW

File: GCRCatalogs/alphaq.py [hunk 1]
```diff
@@ -85,6 +110,42 @@ def native_quantity_getter(native_quantity):
...
+            else:
+                info_dict = dict()
+                for key in fh['galaxyProperties/'+quantity].attrs.keys():
+                    info_dict[key] = fh['galaxyProperties/'+quantity].attrs[key]
+                return info_dict
+    
+    def _get_quantity_info_dict(self, quantity, default=None):
+        return default
+        #TODO needs some fixing
+        print "in get quantity"
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: Note that the code is now required to be 2-3 compatible. Please use print function.

File: GCRCatalogs/alphaq.py [hunk 2]
```diff
@@ -31,8 +44,8 @@ def _subclass_init(self, filename, lightcone=True, **kwargs):
             'dec_true':      (lambda x: x/3600.0, 'dec_true'),
             'redshift':      'redshift',
             'redshift_true': 'redshiftHubble',
-            'disk_sersic_index':'diskSersicIndex',
-            'bulge_sersic_index':'spheroidSersicIndex',
+            'disk_sersic_index':'morphology/diskSersicIndex',
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: At least one space after colon.

File: GCRCatalogs/alphaq.py [hunk 3]
```diff
@@ -85,6 +110,42 @@ def native_quantity_getter(native_quantity):
                 return fh['galaxyProperties/{}'.format(native_quantity)].value
             yield native_quantity_getter
 
+    def _get_native_quantity_info_dict(self,quantity,default=None):
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: Exactly one space after comma.

File: GCRCatalogs/alphaq.py [hunk 4]
```diff
@@ -85,6 +110,42 @@ def native_quantity_getter(native_quantity):
...
+        with h5py.File(self._file,'r') as fh:
+            if 'galaxyProperties/'+quantity not in fh:
+                return default
+            else:
+                info_dict = dict()
+                for key in fh['galaxyProperties/'+quantity].attrs.keys():
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: `for key in fh['galaxyProperties/'+quantity].attrs:`

**Comment by yymao** [comment]
@dkorytov Thanks. I added some inline comments.

Also, along our discussion in #26, can you add version to the yaml config file? For example
```yaml
version: "2.1"
```
and then in the reader, add
```python
if (majorVersion or minorVersion) and not kwargs.get('version', '').startswith('.'.join((majorVersion, minorVersion))):
    raise ValueError('Catalog file version {}.{} does not match config version {}'.format(majorVersion, minorVersion, kwargs.get('version', '')))
```

**Comment by dkorytov** [comment]
Okay. If understand git correctly: I've made all the changes into my fork which should transfer to this pull request...Right?

**Comment by dkorytov** [comment]
Does protoDC2 v2.0 yaml file also need a version number in it?

**Comment by yymao** [comment]
@dkorytov yes, I saw your changes. Let's put a version number to 2.0 too. 

We talked about dropping version number on the filename. Do you want to rename `proto-dc2_v2.1.yaml` to `proto-dc2.yaml` or `protoDC2.yaml`? (Should keep 2.0 the same though.)

File: GCRCatalogs/alphaq.py [hunk 5]
```diff
@@ -21,33 +21,62 @@ def _subclass_init(self, filename, lightcone=True, **kwargs):
 
         assert os.path.isfile(filename), 'Catalog file {} does not exist'.format(filename)
         self._file = filename
-        self.lightcone = lightcone
+        majorVersion = 0
+        minorVersion = 0
+        if (majorVersion or minorVersion) and not kwargs.get('version', '').startswith('.'.join((majorVersion, minorVersion))):
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: @dkorytov Line 26-27 need to be moved to after Line 38.

Comment (dkorytov) [reply GCRCatalogs/alphaq.py]: That's right...:P

**Comment by dkorytov** [comment]
So would protoDC2.yaml file point to the most recent version?

**Comment by yymao** [comment]
@dkorytov I think so. And I can add a routine in `load_catalog` to check this github repo and notify the users if their local `protoDC2.yaml` version differ from what's on the master branch. Does that sound good?

**Comment by yymao** [comment]
@dkorytov I think this PR is now ready to be merged. Have you uploaded the v2.1 file to NERSC?

**Comment by dkorytov** [comment]
That sounds like a good idea to have a version check through github. We will leave the different version yaml files though in case someone wants to look at past versions. Right?

The catalog is ready, but I still don't have write permission to the repo. Eve will be uploading it. I need to get my permissions sorted out.

**Comment by yymao** [comment]
OK @dkorytov, let me know when the catalog is ready on NERSC and I'll merge this PR.

BTW, I implemented version checking in #30. Take a look. Thanks!

File: GCRCatalogs/catalog_configs/proto-dc2_v2.1.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+subclass_name: AlphaQGalaxyCatalog
```
Comment (yymao) [inline GCRCatalogs/catalog_configs/proto-dc2_v2.1.yaml]: @dkorytov please change this line to `subclass_name: alphaq.AlphaQGalaxyCatalog`

File: GCRCatalogs/catalog_configs/proto-dc2_v2.1.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+lightcone: true
+
+version: "2.1"
+
+creators: ['Eve Kovacs', 'Danila Korytov', 'Katrin Heitmann', 'Andrew Benson']
+
+description: |
+          ProtoDC2 is a down-scaled version of the catalog to be generated for LSST-DESC DC2.
+          For a description of the catalog and the methods, please see 
+          https://confluence.slac.stanford.edu/display/LSSTDESC/Extragalactic+Catalog+Development+for+DC2?preview=%2F227167079%2F228300698%2Fdc2-prototype-catalog_release_2.0.pdf
```
Comment (yymao) [inline GCRCatalogs/catalog_configs/proto-dc2_v2.1.yaml]: please use short URL https://goo.gl/fXDQwP

**Comment by yymao** [comment]
@dkorytov you can ignore my comments in this PR, as they are fixed and merged in #31

**Pull request merged**